Include/internal/pycore_frame.h

```diff
@@ -190,11 +190,16 @@ _PyFrame_FastToLocalsWithError(_PyInterpreterFrame *frame);
 void
 _PyFrame_LocalsToFast(_PyInterpreterFrame *frame, int clear);
 
-
 static inline bool
 _PyThreadState_HasStackSpace(PyThreadState *tstate, int size)
 {
-    return tstate->datastack_top + size < tstate->datastack_limit;
+    assert(
+        (tstate->datastack_top == NULL && tstate->datastack_limit == NULL)
```

Comment:
Thanks for adding this!

Btw, you could also have written this check as
```c
assert(
    (tstate->datastack_top == NULL) == (tstate->datastack_limit == NULL)
);
```

But I think your version has the benefit of clarity.  And being a bit verbose in an assert is probably a good thing.